massgov/mayflower-patternlab
# DP-1554 social media links / icons - a11y

This PR adds visually hidden link text to social media links so that assistive technology can provide context for the link target. (Previously, the only link content was the icon) . It also tells assistive tech to ignore the social media icons.

Ticket: https://jira.state.ma.us/browse/DP-1554

Relates to massgov/mass PR:  https://github.com/massgov/mass/pull/403

## To Test

1. Pull branch down
2. move into styleguide `cd styleguide`
3. build mayflower `gulp`
4. browse to any page pattern
5. open dev tools, inspect the link and verify that you can see link text semantically (and that markup matches the screenshot below), close dev tools
7. Scroll down and select the text of one of the footer links (to bring focus down there)
6. Turn on screen reader (make sure your volume is not muted)
  - macos: https://www.apple.com/voiceover/info/guide/_1131.html
  - windows: (if you don't have a jaws license), try chrome vox: https://chrome.google.com/webstore/detail/chromevox/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en
8. Press TAB to move through the footer links, the the footer logo, and finally to the social media links menu
9. Confirm that the link text is announced: Follow us on ________

## Screenshots

### Expanded social media menu link and svg markup
<img width="925" alt="dp-1554-social-media-menu-links" src="https://cloud.githubusercontent.com/assets/3279883/22603532/cf957538-ea15-11e6-8e82-9323f52ddeaa.png">
